Submitted by:            Xi Ruoyao <xry111@mengyan1223.wang>
Date:                    2021-10-07
Initial Package Version: 78.15.0
Upstream Status:         Unknown
Origin:                  Self
Descrption:              Allow to build Spidermonkey with Python-3.10

diff --color -Naur firefox-78.15.0/python/mozbuild/mozbuild/backend/configenvironment.py firefox-78.15.0-fixed/python/mozbuild/mozbuild/backend/configenvironment.py
--- firefox-78.15.0/python/mozbuild/mozbuild/backend/configenvironment.py	2021-09-28 18:04:57.000000000 +0800
+++ firefox-78.15.0-fixed/python/mozbuild/mozbuild/backend/configenvironment.py	2021-10-07 01:45:16.441679948 +0800
@@ -9,7 +9,8 @@
 import sys
 import json
 
-from collections import Iterable, OrderedDict
+from collections import OrderedDict
+from collections.abc import Iterable
 from types import ModuleType
 
 import mozpack.path as mozpath
diff --color -Naur firefox-78.15.0/python/mozbuild/mozbuild/makeutil.py firefox-78.15.0-fixed/python/mozbuild/mozbuild/makeutil.py
--- firefox-78.15.0/python/mozbuild/mozbuild/makeutil.py	2021-09-28 18:04:57.000000000 +0800
+++ firefox-78.15.0-fixed/python/mozbuild/mozbuild/makeutil.py	2021-10-07 01:45:49.555885651 +0800
@@ -7,7 +7,7 @@
 import os
 import re
 import six
-from collections import Iterable
+from collections.abc import Iterable
 
 
 class Makefile(object):
diff --color -Naur firefox-78.15.0/python/mozbuild/mozbuild/util.py firefox-78.15.0-fixed/python/mozbuild/mozbuild/util.py
--- firefox-78.15.0/python/mozbuild/mozbuild/util.py	2021-09-28 18:04:57.000000000 +0800
+++ firefox-78.15.0-fixed/python/mozbuild/mozbuild/util.py	2021-10-07 01:44:29.252352734 +0800
@@ -782,7 +782,7 @@
         self._strings = StrictOrderingOnAppendList()
         self._children = {}
 
-    class StringListAdaptor(collections.Sequence):
+    class StringListAdaptor(collections.abc.Sequence):
         def __init__(self, hsl):
             self._hsl = hsl
 
diff --color -Naur firefox-78.15.0/testing/mozbase/manifestparser/manifestparser/filters.py firefox-78.15.0-fixed/testing/mozbase/manifestparser/manifestparser/filters.py
--- firefox-78.15.0/testing/mozbase/manifestparser/manifestparser/filters.py	2021-09-28 18:04:57.000000000 +0800
+++ firefox-78.15.0-fixed/testing/mozbase/manifestparser/manifestparser/filters.py	2021-10-07 01:45:37.034092219 +0800
@@ -12,7 +12,8 @@
 
 import itertools
 import os
-from collections import defaultdict, MutableSequence
+from collections import defaultdict
+from collections.abc import MutableSequence
 
 import six
 from six import string_types
